20210614/solution.test.js

```diff
@@ -0,0 +1,141 @@
+test('test is working', () => {
+  expect(1 + 1).toBe(2);
+});
+
+// ## 구하는 것
+// 터트려져 사라진 인형의 개수
+// ## 주어진 것
+// - 게임화면 격자 상태 - board
+// - 크레인을 작동시킨 위치 배열 - moves
+// ## 조건
+// 크레인을 작동시킨 위치 -> board 의 열 번호 +1
+// 게임화면 격자 크기 -> N*N 정사각형
+// 인형이 바구니(스택)에 담길 때  같은 모양의 인형이 두개가 연속되면 터뜨려진다.
+
+test('test is working', () => {
+  function solution(board, moves) {
+    const answer = 0;
+    return answer;
+  }~~expect(1 + 1).toBe(2);
+});
+
+function pick(board, move) {
+  const column = board.map((row) => row[move - 1]);
+  const dollIndex = column.findIndex((doll) => doll !== 0);
+  if (dollIndex < 0) {
+    return null;
+  }
+  const doll = board[dollIndex][move - 1];
+  board[dollIndex][move - 1] = 0;
+  return doll;
+}
+
+test('pick at the same column', () => {
+  const board = [
+    [0, 0, 0, 0, 0],
+    [0, 0, 1, 0, 3],
+    [0, 2, 5, 0, 1],
+    [4, 2, 4, 4, 2],
+    [3, 5, 1, 3, 1],
+  ];
+
+  expect(pick(board, 1)).toBe(4);
+  expect(pick(board, 1)).toBe(3);
```

Comment:
같은 위치라는 것을 변수로 드러내도 좋지 않을까요?
```suggestion
  const column = 1;
  
  expect(pick(board, column)).toBe(4);
  expect(pick(board, column)).toBe(3);
```